module/VuFind/src/VuFind/RecordTab/CollectionList.php

```diff
@@ -126,6 +147,12 @@ public function getResults()
     {
         if (null === $this->results) {
             $driver = $this->getRecordDriver();
+            if ($this->showFullHierarchy) {
```

Comment:
If we need to do special processing based on this logic, might it make more sense to do it directly in [initFromRecordDriver](https://github.com/vufind-org/vufind/blob/cdb1273aaea1d449f2f47cfb884458f2bafe2377/module/VuFind/src/VuFind/Search/SolrCollection/Params.php#L68) instead of manipulating things here? It looks like you might possibly just want to change the way collectionID is set in order to achieve the desired effect.